src/subscription-user/dto/get-subscription-user.dto.ts

```diff
@@ -0,0 +1,46 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Expose, Type } from 'class-transformer';
+import { IsDefined, IsNotEmpty, IsOptional, IsString } from 'class-validator';
+
+@Expose()
+export class GetSubscriptionUserDto {
+  @ApiProperty()
+  @IsString()
+  @IsDefined()
+  @IsNotEmpty()
+  id: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsDefined()
+  @IsNotEmpty()
+  subscriptionId: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsDefined()
+  @IsNotEmpty()
+  @Expose()
+  subscriptionName: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsDefined()
+  @IsNotEmpty()
+  subscriptionInfo: string;
+
+  @ApiProperty()
+  @IsDefined()
+  createdAt: Date;
+
+  @ApiProperty()
+  @IsOptional()
+  cancelledAt: Date | null;
+}
+
+export class GetSubscriptionUsersDto {
+  @ApiProperty({ type: [GetSubscriptionUserDto] })
+  @Type(() => GetSubscriptionUserDto)
+  @Expose()
+  subscriptions: GetSubscriptionUserDto[];
```

Comment:
Thanks for this. I don't think we need to nest the subscriptions in an object. I reckon we could just have it as an Array<GetSubscriptionUserDto>

Reply:
I think I need more clarification. Here is a sample of a return for my user who is subscribed to whatsapp:

```
[
    {
        "id": "2762de3b-e2f2-43e3-85c8-7718b3d4f017",
        "subscriptionId": "3b592ff4-97ec-420e-b2fb-b06c460b2b5b",
        "subscriptionName": "whatsapp",
        "subscriptionInfo": "708889927",
        "createdAt": "2024-08-10T22:48:06.554Z",
        "cancelledAt": null
    }
] 
```

Can I just delete the `GetSubscriptionUsersDto` and just map and return in the controller? I have abstracted the serialization to the utils. I think iI am stuck on how to define the `GetSubscriptionUsersDto` because it needs a properfty name.

Reply:
Yes you can!

Reply:
@Tim-Quattrochi sorry again for the delay - both kylee and I were off normal work last week. I think I can see the confusion. I didn't realise it needed a property name! Just map the GetSubscriptionUserDto then. I think that makes sense!